---
title             : "Treasuring Waste: Predicting Covid Cases From Sewage"
shorttitle        : "PREDICTING COVID CASES FROM SEWAGE"
numbersections    : true
fontsize: 12pt

author: 
  - name          : "Jannick Akkermans"
    affiliation   : "1"
    corresponding : yes    # Define only one corresponding author
    email         : "my@email.com"
  - name          : "Sander van Gestel"
    affiliation   : "1"
  - name          : "Lauke Stoel"
    affiliation   : "1"
  - name          : "Annemarie Timmers"
    affiliation   : "1"

affiliation:
  - id            : "1"
    institution   : "Utrecht University"

authornote: |
  Add complete departmental affiliations for each author here. Each new line herein must be indented, like this line.

  Enter author note here.

abstract: |
  One or two sentences providing a **basic introduction** to the field,  comprehensible to a scientist in any discipline.
  
  Two to three sentences of **more detailed background**, comprehensible  to scientists in related disciplines.
  
  One sentence clearly stating the **general problem** being addressed by  this particular study.
  
  One sentence summarizing the main result (with the words "**here we show**" or their equivalent).
  
  Two or three sentences explaining what the **main result** reveals in direct comparison to what was thought to be the case previously, or how the  main result adds to previous knowledge.
  
  One or two sentences to put the results into a more **general context**.
  
  Two or three sentences to provide a **broader perspective**, readily comprehensible to a scientist in any discipline.
  
  <!-- https://tinyurl.com/ybremelq -->
  
keywords          : "keywords"
wordcount         : "X"

bibliography      : ["r-references.bib"]

floatsintext      : yes
figurelist        : no
tablelist         : no
footnotelist      : no
linenumbers       : no
mask              : no
draft             : no
link-citations    : yes

documentclass     : "apa6"
lang              : "english"
classoption       : "man"
output            : 
  papaja::apa6_pdf  : 
    latex_engine        : xelatex
    csl                 : apa6.csl
    nocite              : "@R-base, @R-papaja, @R-ggplot2, @R-dplyr, @R-tidyverse, @datacases, @datawater"
    extra_dependencies  : ["booktabs", "caption", "doi", "float", "graphicx", "lipsum", "natbib", "parskip", "setspace", "threeparttable"]
    
header-includes:
    - \usepackage{setspace}
    - \usepackage{caption}
    - \doublespacing
    - \AtBeginEnvironment{tabular}{\onehalfspacing}
    - \AtBeginEnvironment{tablenotes}{\onehalfspacing}
    - \captionsetup[table]{font={small, stretch = 2}}
    - \captionsetup[figure]{font={small, stretch = 2}}
---

```{r setup, include = FALSE}
library("papaja")
library(dplyr)
library(tidyverse)
r_refs("r-references.bib")
knitr::opts_chunk$set(echo = FALSE, comment = "", cache = TRUE, fig.align = "center")
setwd("~/Universiteit Utrecht/2020-2021/Semester 1/Survey Data Analysis/COVID Assignment")
dat <- readRDS("covid cases waste + tests.RDS")
```

\begingroup
\setlength{\parskip}{2pt}
\raggedbottom
\setcounter{section}{1}

\captionsetup[table]{textfont=it,format=plain,justification=justified, singlelinecheck=false,labelsep=newline,skip=0.5pt}

\captionsetup[figure]{
  font={footnotesize}, 
  labelfont={it},
  justification={raggedright},
  labelsep={period}}

```{r analysis-preferences}
# Seed for random number generation
set.seed(42)
knitr::opts_chunk$set(cache.extra = knitr::rand_seed)
```

```{r data-cleaning, include = FALSE}
# Re-order the data such that those of the same level of measurement are put 
# together. The date_of_report variable has been left out - it is the same as 
# newdate and newdate has a more convenient notation. 
# Rename column names to include a code for the level of measurement: 
# SR = Security Region
# MN = Municipality
# TP = Treatment Plant

dat <- dat %>% rename(SR_code = code, 
                      SR_name = naam,
                      MN_name = Municipality_name, 
                      MN_code = Municipality_code, 
                      MN_province = Province,
                      MN_total.reported = Total_reported, 
                      MN_hospital.admisstions = Hospital_admission, 
                      MN_deceased = Deceased, 
                      MN_pop = inhabitants, 
                      MN_popperkm2 = perkm.2, 
                      MN_sizekm2 = size,
                      TP_code = RWZI_AWZI_code, 
                      TP_name = RWZI_AWZI_name,
                      TP_Xco = X_coordinate,
                      TP_Yco = Y_coordinate, 
                      TP_postal.code = Postal_code, 
                      TP_security.region = Security_region_name, 
                      TP_percentage.in.SR = Percentage_in_security_region,
                      TP_RNA.per.ml = RNA_per_ml, 
                      TP_RNA.flow.per.100000 = RNA_flow_per_100000, 
                      TP_representative.measurement = 
                      Representative_measurement)

dat <- dat %>% select(newdate, SR_code, SR_name, MN_code, MN_name, MN_code:TP_RNA.per.ml, TP_RNA.flow.per.100000, TP_representative.measurement)

str(dat)

# Some variables are reported as character strings, that should be numeric 
# variables - such as MN_sizekm2, TP_percentage.in.SR and TP_RNA.flow.per.100000. 
# This is because the Dutch notation for decimal numbers is comma's, where R would 
# expect dots, and vice versa. Other variables are noted in the wrong order of size # because of this. 
# Replace comma's by dots for all variables with decimals

dat$MN_sizekm2 <- gsub(",", ".", dat$MN_sizekm2)
dat$MN_popperkm2 <- gsub(",", ".", dat$MN_popperkm2)
dat$TP_percentage.in.SR <- gsub(",", ".", dat$TP_percentage.in.SR)
dat$TP_RNA.flow.per.100000 <- gsub(",", ".", dat$TP_RNA.flow.per.100000)

# Multiply by 1000 to get rid of the "." in the population variable
dat <- dat %>% mutate(MN_pop = MN_pop*1000)

# Redefine variables as numeric:
dat$TP_Yco <- as.numeric(dat$TP_Yco)
dat$MN_popperkm2 <- as.numeric(dat$MN_popperkm2)
dat$MN_sizekm2 <- as.numeric(dat$MN_sizekm2)
dat$TP_RNA.flow.per.100000 <- as.numeric(dat$TP_RNA.flow.per.100000)
dat$TP_percentage.in.SR <- as.numeric(dat$TP_percentage.in.SR)

# Check the structure again
str(dat)
```

In December 2019, a virus known as SARS-Cov-2 (COVID-19) was initiated in Wuhan, China. This variant of the SARS coronavirus, which shocked the world in 2003, caused a worldwide pandemic with many consequences. However, this virus is more dangerous since 20 to 40% of the patients show no symptoms, contributing to the silent spread of the virus [@vallejo2020highly].

Although patients do not always show symptoms of the virus, they do leave RNA particles of the virus in their feces as shown by for example @pan2020viral. The virus can sustain itself for a long period of time within the feces, in some cases even one or more months after the respective patient has tested negative for RNA particles in their feces [@vallejo2020highly]. Therefore, the amount of RNA particles could be an indicator of the true number of COVID-19 patients within a country or security region. Given this information, this project attempted to build a model that estimates the true number of new positive cases in a security region on a given day based on the RNA flow seven days prior.

Ever since the start of the pandemic in the Netherlands, the National Institute for Public Health and the Environment (RIVM) has been collecting samples from sewage treatment plants (STPs) and testing them for RNA presence. The research started small, with only 29 out of the 355 STPs in April, but since the beginning of September 2020, all STPs in the Netherlands are sampled once or multiple times a week. 

After the samples are taken, they are transported at a controlled temperature to the RIVM, where they are analysed by researchers for RNA particles. RNA is isolated and Polymerase Chain Reaction (PCR) is performed on the samples to determine the amount of RNA particles present in the wastewater [@Riool].

Multiple equations result in an estimate of the number of RNA particles per 100,000 inhabitants of the Netherlands, which was made possible by mapping the number of households connected to a STP. In these equations, the RIVM corrects for the amount of water that flows into the STPs [@Rijksoverheid]. This is needed, because when it has rained, this results in more water in the sewage. More water would lower the concentration of RNA particles in the wastewater, hereby possibly distorting the estimate. The resulting numbers are presented on the Corona Dashboard [@Dashboard]. 

The remainder of this report is structured as follows. Section \ref{sec:litrev} will discuss prior research in the field of measuring RNA. Section \ref{sec:datmet} describes the data and the analyses conducted in this research. The results of those analyses are described in Section \ref{sec:results}. Section \ref{sec:conclusion} interprets those results and subsequently provides a conclusion. Finally, Section \ref{sec:discussion} discusses certain things that could have gone better in this research.  

\section{Literature Review}
\label{sec:litrev}

As the aim of this project is to estimate the true number of positive corona tests in a security region from the RNA in the wastewater, literature regarding RNA as a predictor of positive cases and PCR, by which RNA estimates are retrieved, should be reviewed. The following section therefore dives deeper into the usefulness of RNA particles as a predictor of corona cases and the method by which the number of particles is determined.

\subsection{Relevance of measuring RNA in sewage water}
Early on in the pandemic, the RIVM began sampling wastewater. One of the reasons to do this, was the use of wastewater samples to detect diseases in the past. For example, wastewater has been used to detect and monitor the spread of polio since the 1980s, with the World Health Organization (WHO) issuing guidelines to do this [@CDC; @mao2020potential]. 
Moreover, RNA was found to be present in the feces of both symptomatic and asymptomatic carriers, hereby being able to capture both types of infections [@randazzo2020sars]. Because of this, RNA can serve as an indicator for rises and falls in infections, independent of the number of positive tests. In addition, as RNA is measured locally, it can not only serve as a national, but also a regional indicator of the total number of infections. Altogether, wastewater sampling provides some valuable opportunities to improve the detection and monitoring of the spread of COVID-19.

\subsection{The PCR method}
As we have just established that measuring RNA is important, it is also important to understand the method by which the measuring is done. As mentioned in the introduction, sewage water samples are transported at a controlled temperature to the RIVM. There, researchers isolate the RNA of the virus and perform a technique called PCR in order to determine the number of RNA in the sewage water. 

PCR follows a three step process which consists of (1) denaturation of double-stranded DNA, (2) annealing of primers, and (3) primer extension [@schochetman1988polymerase]. Denaturation of double-stranded DNA involves separating the two strands that together form the complex DNA-sequence. In the next step, primers are added to both strands. A primer is a single strand that is complementary to the DNA-sequence it is attached to. The new combinations of DNA strands are then synthesized together to form new DNA sequences. This process is iterated until there is not enough primer left to form new sequences. Afterwards, the amount of DNA is measured to provide the total amount of DNA in a given sample.

The amount of RNA is determined through an adaptation of this technique, called Reverse Transcription-Polymerase Chain Reaction (RT-PCR). The procedure is as follows according to @rio2014reverse: first, a primer is added to the RNA strand. This new synthesized RNA-DNA combination is then used as a template for Reverse Transcriptase, in which a single-stranded cDNA copy is created. However, this cDNA strand is only a proportion of the original RNA strand. The newly created strand is finally used in the PCR method to determine the amount of RNA particles in the population (i.e. the sewage water of a security region in the Netherlands).

Although the PCR method can produce results relatively fast according to @garibyan2013research, it does have a few limitations. First of all, PCR is a highly sensitive technique, so any contamination of the sample can lead to misleading results. Secondly, PCR depends on the addition of a primer to create new DNA sequences. The drawback in this case is that the creation of the primer requires prior knowledge of the target sequence you are attempting to create [@garibyan2013research]. Therefore, PCR can only be applied to known pathogens or genes. Nevertheless, this method has its advantages and is currently used by the RIVM.

\subsection{RNA as a predictor for COVID infections}
Once the number of RNA has been measured, it could be used as a predictor for the number of COVID infections. However, there is something that needs to be taken into account when using this predictor. @peccia2020measurement attempted to track the spread of COVID-infections in Connecticut by measuring RNA flow in wastewater. They found that RNA concentrations in wastewater were six to eight days ahead of the corresponding reports of positive tests. Therefore, RNA cannot be compared to the number of positive tests one-on-one, but rather, with a multiple day time-lag. 

Another problem with the current data on RNA in sewage water, is that it is unknown how much RNA particles need to be shed, for them to be detected during testing [@CDC]. This complicates prediction, as some RNA may go undetected, resulting in an underestimate of the total number of infections in the Netherlands as a whole (or a single region, for that matter). 

Finally, when taking a sample of the sewage water, the concentration of RNA that it contains may be less than the original concentration in the sewage water. This may be due to the type of the environment or the physical and chemical properties of the environment [@lahrich2020review]. For example, some RNA particles may die due to the temperature of the wastewater or because of too much sunlight exposure. Since the RNA concentration can be lowered, we need to be careful when using it as a predictor to estimate the number of COVID infections.

\section{Data and Methodology}
\label{sec:datmet}
\subsection{Structure of the data}
We are working with two datasets, both collected by the Dutch National Institute for Public Health and the Environment (RIVM). The first dataset contains the total number of positive tests reported per day per municipality. It also contains information on key characteristics of the municipality, such as population density and which security region it is part of. The second dataset contains data recorded on the level of sewage treatment plants. The key variable here is the average concentration of SARS-CoV-2 RNA measured in the daily amount of sewage water per 100,000 inhabitants. This dataset also contains crucial metadata of the sewage treatment plants, such as in which security region the area of responsibility of this treatment plant falls. The two datasets were matched to each other by the variable in which security region a municipality and a treatment plant’s area fall respectively.

\subsection{Challenges in the data}
Our goal is to estimate the true number of COVID-19 patients at any given day, based on the data we have available describing the RNA flow in the sewage water. To do this, we first have to establish if there is a relationship between the RNA flow and the number of positive tests per day. Given the data structure, we are faced with a few challenges before we can take on this question.

First of all, the data on the number of positive tests are recorded on the level of municipalities, whereas the data on RNA flow are recorded on the treatment plant level. In the most straightforward cases, we can aggregate the RNA flow data to municipality level data by virtue that the datasets were already matched by security region code. See SR1 of Figure \ref{fig:diagram}.

\begin{figure}[!hbt]
\begin{center}
\includegraphics{Diagram Bijgesneden.png}
\begin{tiny}
\caption{Illustration depicting the structure of the data.}
\label{fig:diagram}
\end{tiny}
\end{center}
\end{figure}

However, some sewage treatment plants also treat water from outside their primary security region, creating double entries in the dataset and making simple matching impossible. Luckily, the dataset also provides information on what percentage of the water a treatment plant processes comes from which security region, so we can weight the RNA flow by this variable. See SR2 of Figure \ref{fig:diagram}.
Furthermore, some very large municipalities produce so much sewage water in one day, that multiple treatment plants are required to process it, causing a second kind of double entries. See SR3 of Figure \ref{fig:diagram}. Unfortunately, there are no data available that specify how much of the water from these large municipalities goes to which treatment plant, making it impossible to establish if there is a relationship between the RNA flow and the number of positive cases on the municipality level.

We take the above relationships into account in our analyses. Furthermore, we conclude that we have to aggregate our data to the Security Region level when establishing a relationship between RNA flow and number of positive cases, to get an accurate indicator. Lastly, the data we use in our analyses are a subset of the original dataset, because the dataset only includes all sewage treatment plants from the 7th of September onwards. 

We work towards our goal through asking and answering several sub-questions. The following section contains the analysis for answering the following questions:
\begin{APAenumerate}
  \setlength{\itemsep}{0pt}
  \setlength{\parskip}{0pt}
   \item How does the total number of reported infections in the Netherlands as a whole develop over time? (Section \ref{sec:sub1})
   \item What does this trend look like displayed per 100,000 inhabitants of the Netherlands? (Section \ref{sec:sub2})
   \item What is the mean level of RNA particles found in the water per 100,000 inhabitants?
   \item What is the relationship between RNA particles and the total number of infections on one given moment?
   \item How does this relationship look over time?
\end{APAenumerate}

\subsection{How does the total number of reported infections in the Netherlands as a whole develop over time?}
\label{sec:sub1}

In order to answer this question we need a function that calculates the number of total reported infections per day. We wrote a function to aggregate the data by the number of reported infections in a municipality on a given day to the total number of reported infections on that day in the Netherlands as a whole, discarding double entries per municipality. The resulting output is visualized in Figure \ref{fig:2a}, where you can see the total number of reported infections we have had over time in the Netherlands. The second wave is also clearly visible, where you see the total number of reported infections increasing more rapidly in October compared to September. 

We are also interested in the daily fluctuations of the number of reported infections in the Netherlands. To show this, we altered the function such that it would calculate the difference in total reported cases between each day and the day before. The results are visualized in Figure \ref{fig:2b}. We see that the peak of new infections in the second wave is somewhere in the end of October, as we would expect from the previous figure.  


```{r fig.width = 10, include = FALSE}
#create dataframe with only the observations from 7th of September onward
all_dates  <- unique(dat$newdate) #243 recorded dates, starting March 13th
used_dates <- dat %>% filter(newdate %in% all_dates[-c(1:178)]) #dates from 7th of September

#this function calculates the total of reported cases on one day, across all municipalities
totalCalculator <- function(used_dates) {
  dates <- unique(used_dates$newdate) #create a list of unique dates in the dataset
  totals <- numeric(length(dates)) #make an object with the length of the number of dates
  
  #subtract nr of reported cases of the day before each day to get the difference
  for (i in 1:(length(totals))) {
    step1 <- filter(used_dates, newdate == dates[i]) #select date we want to calculate the nr of new cases for
    
    #create data frame of the  dates, and remove duplicates (total reported per municipalities are the same across duplicates)
    df <- step1[!duplicated(step1[,"MN_name"]),] 
    
    #calculate the sum of all new reported cases per municipality and store in object
    sums <- sum(df$MN_total.reported, na.rm = TRUE)
    totals[i] <- sums
  }
  
  #create a data frame with all the outcomes, matched with each date
  totals_frame <- as.data.frame(cbind(dates, totals))
  colnames(totals_frame) <- c("Date", "Total number of reported infections") #rename columns
  totals_frame$`Total number of reported infections` <- as.numeric(totals_frame$`Total number of reported infections`) #store the totals as numeric values.
  
  #create a plot of the totals over time
  plot <- barplot(totals_frame$`Total number of reported infections`, 
                  col  = "greenyellow",
                  xlab = "Date", 
                  ylab = "Reported infections", 
                  ylim = c(0,400000), 
                  names.arg = as.Date(dates))
  
  #Output a data frame with the total of new cases per day
  output = list(totals_frame, plot)
  return(output)
}

```

```{r 2a, results = 'hide', fig.align = "center", fig.cap = x1, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x1 <- paste(
    "Cumulative number of reported infections per day in the Netherlands over time.")
a2 <- totalCalculator(used_dates)

```

```{r fig.width = 10, include = FALSE}
#this function calculates the difference in reported cases between one day and the day before that:
differenceCalculator <- function(used_dates) {
  dates <- unique(used_dates$newdate) #create a list of unique dates in the dataset
  differences <- numeric(length(dates)-1) #make an object with the length of one fewer than the number of        dates. Why one fewer? We cannot calculate the number of new reported cases of the first day, because the number of cases from the preceding day is unknown.  
  
  #subtract nr of reported cases of the day before each day to get the difference
  for (i in 1:(length(differences))) {
    step1 <- filter(used_dates, newdate == dates[i]) #select date of reference
    step2 <- filter(used_dates, newdate == dates[i+1]) #select date we want to calculate the nr of new cases for
    
    #create data frames of both sets of dates, and remove duplicates (total reported per municipalities are the same across duplicates)
    df1 <- step1[!duplicated(step1[,"MN_name"]),] 
    df2 <- step2[!duplicated(step2[,"MN_name"]),]
    
    #calculate the difference between the two dataframes for each set of dates and store in object
    diff <- (sum(df2$MN_total.reported, na.rm = TRUE) - sum(df1$MN_total.reported, na.rm = TRUE))
    differences[i] <- diff
  }
  
  #create a data frame with all the outcomes, matched with each date
  diff_frame <- as.data.frame(cbind(dates[-1], differences))
  colnames(diff_frame) <- c("Date", "Difference in reported infections") #rename columns
  diff_frame$`Difference in reported infections` <- as.numeric(diff_frame$`Difference in reported infections`) #store the differences as numeric values.
  
  
  #create a plot of the totals over time
  plot <- barplot(diff_frame$`Difference in reported infections`, 
                  col = "greenyellow", 
                  xlab = "Date", 
                  ylab = "New reported infections", 
                  names.arg = as.Date(dates[-1]))
  
  #output a data frame with new cases per day
  output = list(diff_frame, plot)
  return(output)
}


```

```{r 2b, results = 'hide', fig.align = "center", fig.cap = x2, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x2 <- paste(
    "Number of new reported infections per day in the Netherlands over time")
b2 <- differenceCalculator(used_dates)
```

\subsection{What does this trend look like displayed per 100,000 inhabitants of the Netherlands?}
\label{sec:sub2}

Next, we visualized the number of reported cases per 100,000 inhabitants of the Netherlands. Again, we decided to do this for both the total number of infections we have had over time, as well as the daily differences. As you can see in the figures below, the shape of the distribution is the same, but the scale on the y-axis is different. This is expected, as we took the functions of the previous step and multiplied the total number of reported infections by the number of inhabitants of the municipality divided by 100,000.

\subsection{What is the mean level of RNA particles found in the water per 100,000 inhabitants?}

We saw that the number of infections had been increasing since the beginning of September and peaked in October. As one only requests a test after experiencing symptoms, one would expect the amount of RNA particles in the sewage water to precede the increase in the number of infections. Therefore, we have visualized the distribution of the RNA particles in the Netherlands over time. In the figure below, the mean number of RNA particles per 100,000 inhabitants summed over all sewage water installations for every day is plotted. The shape of the plot loosely follows the same trend as the plot of the reported infections per day, but its shape is much less smooth. 

```{r fig.width = 10, include = FALSE}
#in the data preparation, we already multiplied the municipality population by 1000,
#so no need to do that again.
#however, we do need make a new variable containing the number of cases per 100.000 people
#to do that, we multiple the total number of reported cases by 100.000 divided by the municipality population and give it an informative name.
used_dates$cases_per_100000 <- used_dates$MN_total.reported * (100000/used_dates$MN_pop)

#make a function that calculates the mean number of infection per 100.000 inhabitants per day.
#so, this is over the Netherlands as a whole.
totalCalculator2 <- function(used_dates) {
  used_dates$cases_per_100000 <- used_dates$MN_total.reported * (100000/used_dates$MN_pop)
  dates <- unique(used_dates$newdate) #create a list of dates
  totals <- numeric(length(dates)-1) #make an object with the length of one fewer than the number of dates. Why one fewer? We cannot calculate the number of new reported cases of the first day, because the number of cases from the preceding day is unknown.
  
  for (i in 1:(length(totals))) {
    step1 <- filter(used_dates, newdate == dates[i]) #select the date we want to calculate the nr of cases per 100.000 inhabitants for
    
    #create data frame of the  dates, and remove duplicates (total reported per municipalities are the same across duplicates)
    df <- step1[!duplicated(step1[,"MN_name"]),] 
    
    #calculate the mean number of reported cases per 100.000 inhabitants in the Netherlands and store in object
    means <- mean(df$cases_per_100000, na.rm = TRUE)
    totals[i] <- means #give it a name
  }
  
  #create a dataframe with the totals by the respecitve dates
  tot_frame <- as.data.frame(cbind(dates[-1], totals))
  #give the columns informative names
  colnames(tot_frame) <- c("Date", "Reported infections per 100.000")
  #and make the increase numeric
  tot_frame$`Reported infections per 100.000` <- as.numeric(tot_frame$`Reported infections per 100.000`)
  
  #create a plot of the increase per 100.000 inhabitants over time
  barplot(tot_frame$`Reported infections per 100.000`, 
          col = "greenyellow",
          xlab = "Date", 
          ylab = "Reported infections per 100,000 inhabitants", 
          names.arg = as.Date(dates[-1]))
  
  return(tot_frame)
}

```

```{r 3a, results = 'hide', fig.align = "center", fig.cap = x3, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x3 <- paste(
    "Cumulative number of reported infections per day per 100,000 inhabitants in the Netherlands over time")
totalCalculator2(used_dates)
```

```{r fig.width = 10, include = FALSE}
#in the data preparation, we already multplied the municipality population by 1000,
#so no need to do that again.
#however, we do need make a new variable containing the number of cases per 100.000 people
#to do that, we multiple the total number of reported cases by 100.000 divided by the municipality population and give it an informative name.
used_dates$cases_per_100000 <- used_dates$MN_total.reported * (100000/used_dates$MN_pop)

#make a function that calculates the mean number of infection per 100.000 inhabitants per day.
#so, this is over the Netherlands as a whole.

differenceCalculator2 <- function(used_dates) {
  used_dates$cases_per_100000 <- used_dates$MN_total.reported * (100000/used_dates$MN_pop)
  dates <- unique(used_dates$newdate) #create a list of dates
  differences <- numeric(length(dates)-1) #make an object with the length of one fewer than the number of dates. Why one fewer? We cannot calculate the number of new reported cases of the first day, because the number of cases from the preceding day is unknown.
  for (i in 1:(length(differences))) {
    step1 <- filter(used_dates, newdate == dates[i]) #select date of reference
    step2 <- filter(used_dates, newdate == dates[i+1]) #select date we want to calculate the nr of new cases for
    
    #create data frames of both sets of dates, and remove duplicates (total reported per municipalities are the same across duplicates)
    df1 <- step1[!duplicated(step1[,"MN_name"]),]
    df2 <- step2[!duplicated(step2[,"MN_name"]),]
    
    #calculate the differences by subtracting the previous day from the one the for loop is on then
    diff <- (mean(df2$cases_per_100000, na.rm = TRUE) - mean(df1$cases_per_100000, na.rm = TRUE))
    #and store it in our differences vector
    differences[i] <- diff
  }
  
  #create a dataframe with the differences by the respecitve dates
  diff_frame <- as.data.frame(cbind(dates[-1], differences))
  #give the columns informative names
  colnames(diff_frame) <- c("Date", "Increase in reported infections per 100.000")
  #and make the increase numeric
  diff_frame$`Increase in reported infections per 100.000` <- as.numeric(diff_frame$`Increase in reported infections per 100.000`)
  
  #create a plot of the increase per 100.000 inhabitants over time
  barplot(diff_frame$`Increase in reported infections per 100.000`, 
          col = "greenyellow",
          xlab = "Date", 
          ylab = "New reported infections per 100,000 inhabitants", 
          names.arg = as.Date(dates[-1]))
  
  return(diff_frame)
}

```

```{r 3b, results = 'hide', fig.align = "center", fig.cap = x4, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x4 <- paste(
    "Number of new reported infections per day per 100,000 inhabitants in the Netherlands over time")
differenceCalculator2(used_dates)
```

```{r fig.width=10, echo = FALSE}

used_dates$TP_RNA.flow.per.100000 <- as.numeric(used_dates$TP_RNA.flow.per.100000)

virusCalculator_new <- function(used_dates) {
  #creates a vector containing the unique values of the dates column of data. So vector with dates from september the seventh
  #till october the tenth
  dates <- unique(used_dates$newdate)
  #makes an empty numeric vector which length is equal to number of dates
  means <- numeric(length(dates))
  for (i in 1:length(means)) {
    #puts all the data of a particular date into date_data. the particular date changes at each iteration.
    date_data <- filter(used_dates, newdate == dates[i])
    #delete rows if the water installation name is already in other row.
    df1 <- date_data[!duplicated(date_data[,c("SR_name","TP_code")]),]
    #take the mean of the rna per 100000 column and store in means vector
    means[i] <- mean(df1$TP_RNA.flow.per.100000, na.rm = TRUE)
  }
  #put date and means vector in dataframe.
  means_frame <- as.data.frame(cbind(dates, means))
  #define column names.
  colnames(means_frame) <- c("Date", "Mean RNA flow per 100000")
  #coerce the column into double precision vector.
  means_frame$`Mean RNA flow per 100000` <- as.double(means_frame$`Mean RNA flow per 100000`)
  #create a barplot 
  barplot(means_frame$`Mean RNA flow per 100000`, col='greenyellow', 
          xlab = 'Date', ylab = 'Mean RNA flow per 100,000 inhabitants', names.arg = dates)
  return(means_frame)
}

```

```{r 4, results = 'hide', fig.align = "center", fig.cap = x5, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x5 <- paste(
    "Mean RNA flow per 100,000 inhabitants in the Netherlands per day")
virusCalculator_new(used_dates)
```

The reason for the different shapes of the plot showing the mean RNA flow and the plot showing the increase in infections could be the fact that not all treatment plants report the RNA flow every day, and that the number of reporting treatment plants differs per day and per region. In order to explore this idea we plotted the number of installations from which a sample was taken over time. The figure below shows that the number of installations from which a sample is taken differs per day. The figure also shows that the maximum number of sewage water installations from which samples are taken lies somewhere around one hundred. Since there are more than three hundred installations from which a sample can be taken [@Riool], there is not a single day where a sample is taken from all installations. So every day, samples are taken from different installations that process waste water from different regions. If every region produced the same amount of RNA particles it would not matter that we do not observe each installation each day. Since we take the mean of the amount of RNA particles, the estimate would not be harmed by a missing installation. However, the figure that shows the mean number of RNA particles observed each day suggests that the estimate of the mean number of RNA particles is not correct. If it were correct, then the values would not fluctuate as much as they do now and would follow the trend of new infections more clearly. Thus, the plot supports our claim that the different shapes of the plot showing the mean RNA flow and the plot showing the increase in infections could be explained by the fact that not all treatment plants report the RNA flow every day, and that the number of reporting treatment plants differs per day and per region.

```{r fig.width = 10, echo = FALSE}
totalCalculator_installations <- function(sewer_data) {
  dates <- unique(sewer_data$newdate) #create a list of unique dates in the data set 
  totals <- numeric(length(dates)) #make an object with the length of the number of dates
  
  #subtract number of reported cases of the day before each day to get the difference
  for (i in 1:(length(totals))) {
    step1 <- filter(sewer_data, newdate == dates[i]) #select date we want to calculate the nr of new cases for
    
    #create data frame of the  dates, and remove duplicates (of installations)
    df <- step1[!duplicated(step1[,"TP_code"]),]
    codes<-df$TP_code[!is.na(df$TP_code)]
    #calculate the number of installations per day
    sums <- length(codes)
    totals[i] <- sums
  }
  barplot(totals, 
          col = 'greenyellow', 
          names.arg = dates, 
          ylab = 'Date', 
          xlab = 'Number')
  
}

```

```{r 5, results = 'hide', fig.align = "center", fig.cap = x6, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x6 <- paste(
    "Number of installations per day")
totalCalculator_installations(used_dates)
```

\subsection{What is the relationship between RNA particles and the total number of infections on one given moment?}

As mentioned in the data description, some installations process water from multiple security regions, rendering the current RNA flow variable unrepresentative. We do have a variable available that contains the proportion of water from the security region processed by that respective installation. We multiplied these two columns to create a representative RNA flow variable.

Now that the RNA flow has been weighted, we want to inspect how the RNA flow is related to the increase in the number of infections on the level of the security regions. To achieve this, we altered the current functions for calculating the RNA flow and increase in number of infections per day so that these values are reported for each security region per day. Subsequently, we compared the RNA flow from one day to the increase in the number of reported infections from seven days later. The reason for this is that there is generally a seven-day time lag between a person contracting COVID – leaving RNA particles in the sewage water – and getting a positive test [@peccia2020measurement]. A scatterplot with the aforementioned data should provide a first impression of how they are related to each other. Since not every security region has data about the RNA flow on any given day and data about the increase in the number of infections seven days later, only security regions that have data for both days are selected. Subsequently, the correlation for those data is calculated.

```{r fig.width = 10, include = FALSE}
rnaWeighting <- function(used_dates) {
  dates <- unique(used_dates$newdate) #create a list of all dates
  adjusted_means <- numeric(length(dates)) #create a vector of zeros that is equal in length to the number of dates
  
  non_na_data <- used_dates[!is.na(used_dates$TP_RNA.flow.per.100000),] #just for now eliminate NA rows
  non_na_data2 <- non_na_data[!is.na(non_na_data$TP_percentage.in.SR),]
  non_na_data2$TP_percentage.in.SR <- as.double(non_na_data2$TP_percentage.in.SR)
  non_na_data2$adjusted_RNA_flow <- (non_na_data2$TP_RNA.flow.per.100000 *    non_na_data2$TP_percentage.in.SR) #adjust rna flow by multiplying it with the proportion of water from that security region processed by that RWZI
  
  for (i in 1:length(adjusted_means)) {
    day_data <- filter(non_na_data2, newdate == dates[i])
    df1 <- day_data[!duplicated(day_data[,c("SR_name","TP_code")]),]
    
    
    adjusted_means[i] <- mean(df1$adjusted_RNA_flow, na.rm = TRUE)
  }
  means_frame <- as.data.frame(cbind(dates, adjusted_means))
  colnames(means_frame) <- c("Date", "Adjusted mean RNA flow per 100000")
  means_frame$`Adjusted mean RNA flow per 100000` <- as.double(means_frame$`Adjusted mean RNA flow per 100000`)
  
     #create a plot of the increase per 100.000 inhabitants over time
   barplot(means_frame$`Adjusted mean RNA flow per 100000`,
                  xlab = "Date", 
                  ylab = "Mean RNA flow per 100.000 inhabitants", 
                  col = "greenyellow",
                  names.arg = as.Date(dates))
  
  return(means_frame)
}

newCalculator <- function(used_dates) {
  sr_frame <- data.frame()
  dates <- unique(used_dates$newdate) #extract a vector of dates
  regions <- unique(used_dates$SR_name) #extract a vector of all security regions
  used_dates$adjusted_RNA_flow <- (used_dates$TP_RNA.flow.per.100000 *  used_dates$TP_percentage.in.SR)
  
  for (i in 1:(length(dates)-1)) {
    step1 <- filter(used_dates, newdate == dates[i]) #filter the data for date 1
    step2 <- filter(used_dates, newdate == dates[i+1]) #filter the data for the day after the current day
  
    for (j in 1:length(regions)) {
      r1_data <- filter(step1, SR_name == regions[j]) #filter all data for a given region on the first day
      r2_data <- filter(step2, SR_name == regions[j]) #filter all data for a given region on the second day
    
      df1 <- r1_data[!duplicated(r1_data[,"MN_name"]),] #delete all duplicate municipality rows
      df2 <- r2_data[!duplicated(r2_data[,"MN_name"]),] #delete all duplicate municipality rows on the second day
    
      increase <- sum(df2$MN_total.reported, na.rm = TRUE) - sum(df1$MN_total.reported, na.rm = TRUE)
      rna_flow <- sum(r2_data$adjusted_RNA_flow)
      r_frame <- data.frame(Date = dates[i+1], Region = regions[j], Increase = increase, RNA.flow = rna_flow)
      sr_frame <- rbind(sr_frame, r_frame)
    }
  }
  non_na_frame <- sr_frame[complete.cases(sr_frame),]
  return(non_na_frame)
}

good_frame <- newCalculator(used_dates)
good_frame

good_dates <- unique(good_frame$Date) #extract a list of the dates in the dataframe
day1 <- filter(good_frame, Date == good_dates[1]) #extract the data of the first day, i.e. 8th of september
day2 <- filter(good_frame, Date == good_dates[8]) #extract the data of increase in infections 7 days later, i.e. 15th of september
regions <- unique(used_dates$SR_name)
day_frame <- data.frame()
for (i in 1:(length(regions))) {
  dat1 <- filter(day1, Region == regions[i])
  dat2 <- filter(day2, Region == regions[i])
  if (nrow(dat1) == 1 && nrow(dat2) == 1) {
    d_frame <- data.frame(Region = regions[i], Reported = dat2$Increase, RNA.flow = dat1$RNA.flow)
    day_frame <- rbind(day_frame, d_frame)
  }
}

```

```{r 6, results = 'hide', fig.align = "center", fig.cap = x7, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x7 <- paste(
    "RNA flow per 100.000 inhabitants in the Netherlands over time")
rnaWeighting(used_dates)
```

```{r 7, results = 'hide', fig.align = "center", fig.cap = x8, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x8 <- paste(
    "Relation between RNA flow and increase in infections for one timestamp")
plot(day_frame$RNA.flow, day_frame$Reported, col = "red", pch = 19, xlab = "RNA flow", ylab = "Increase in number of infections")
     abline(lm(Reported ~ RNA.flow, data = day_frame), lwd = 2)
```


Figure \ref{fig:7} shows the correlation between the RNA flow on the 8th of September and the increase in the number of infections on the 15th of September. As can been seen from the scatterplot, these data do not seem be strongly related to each other. Additionally, the correlation associated with this scatterplot is 0.027, which indicates that these data almost show no relationship to each other. This strikes us as very surprising, since one would expect a relationship between the RNA flow and the increase in the number of infections. In the next step, we extend the analysis conducted over the period of September until November to see if this trend persists.

\subsection{How does this relationship look over time?}
```{r fig.width = 10, include = FALSE}
corCalculator <- function(good_frame, used_dates) {
  dates <- unique(good_frame$Date)
  regions <- unique(used_dates$SR_name)
  cor_frame <- data.frame()
  
  for (i in 1:(length(dates)-7)) {
    day1 <- filter(good_frame, Date == dates[i]) #filter data for the first day
    day2 <- filter(good_frame, Date == dates[i+7]) #filter data for 7 days later
    day_frame <- data.frame()
    observations <- 0
  
    for (j in 1:length(regions)) {
      dat1 <- filter(day1, Region == regions[j])
      dat2 <- filter(day2, Region == regions[j])
      if (nrow(dat1) == 1 && nrow(dat2) == 1) {
        observations <- observations + 1
        d_frame <- data.frame(Region = regions[j], Reported = dat2$Increase, RNA.flow = dat1$RNA.flow)
        day_frame <- rbind(day_frame, d_frame)
      } else {
        next
      }
    }
    cor_f <- data.frame(Date = dates[i+7], Correlation = cor(day_frame$Reported, day_frame$RNA.flow), Observations = observations)
    cor_frame <- rbind(cor_frame, cor_f)
  }
  return(cor_frame)
}

correlation_frame <- corCalculator(good_frame, used_dates)
correlation_frame
mean(correlation_frame$Correlation, na.rm = TRUE)

corr_frame <- correlation_frame[complete.cases(correlation_frame),]
corr_frame$Date <- as.Date(corr_frame$Date)

```

Figure \ref{fig:8} shows that the correlations fluctuate over time, which seems very interesting. If the data points would have formed a straight horizontal line, the estimation of the true number of infections in the Netherlands would have been more straightforward, since it would mean that the relationship between the RNA flow and the increase in infections would have been the same for every day. The mean correlation between the datapoints in this scatterplot is 0.33, which is a moderate positive correlation. This means that as the pandemic progresses, the relationship between the RNA flow and the increase in the number of infections becomes stronger. We have two hypotheses as to why this would be the case: it could be a result of increased consistency and precision in the measurements as we get more rigorous in reporting these data, or – as the number of cases has only increased over the period we analysed – it could mean that RNA flow becomes a better predictor of the total number of reported cases, as both measures have higher values.

```{r 8, results = 'hide', fig.align = "center", fig.cap = x9, fig.height = 4, fig.width = 5.5, fig.pos = "H"}
x9 <- paste(
    "Correlations between RNA flow and increase in infections over time")
plot(x = corr_frame$Date, y = corr_frame$Correlation, col = "red", pch = 19, xlab = "Date", ylab = "Correlation")
abline(lm(Correlation ~ Date, data = corr_frame), lwd = 2)
```

\section{Results}
\label{sec:results}

\section{Conclusion}
\label{sec:conclusion}

As of yet, we are unable to make a reliable estimate of the true number of infected people in the Netherlands. First and foremost, this is due to the unstable nature of the correlation between the RNA flow and the total number of recorded infections. However, the moderate, positive correlation between these two variables indicates that the RNA flow becomes an increasingly better predictor as we get further along in the pandemic. If we do essay to construct a prediction model, we recommend to use only the most recent RNA flow data to build it, as they are likely the most reliable. 

Our next line of investigation will be to try to combine the RNA flow as a predictor with other available variables in our dataset, such as population density. We would also like to separate this analysis by security region, to isolate the effect of geographic location on the predictor. 

If we had infinite time and resources, we would like to enrich our analysis by incorporating other types of data in our analysis. For example, we could use behavioural data on when people are more or less likely to get tested when displaying symptoms, to improve the estimate of the true number of infected people. Another option would be to include environmental data on how much rain fell in each region on each day, to correct for the influence this might have on the concentration of RNA particles in the sewage water. For now, we conclude that the RNA flow is an imperfect indicator of the true number of infected people in the Netherlands, but that it could potentially be very valuable in combination with other data sources. 

\section{Discussion}
\label{sec:discussion}

\endgroup
\newpage

\stepcounter{section}
\section*{\normalfont{References}}

\begingroup
\setlength{\parindent}{-0.5in}
\setlength{\leftskip}{0.5in}
\setlength{\parskip}{2pt}

<div id="refs" custom-style="Bibliography"></div>
\endgroup
